### PR TITLE
fix(build): make npm linking work pt. 2

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY

The recent upgrade to Webpack 5 in #16701 broke npm linking. While #16867 added watching for changed files, it still didn't update the browser with the new assets. This switches from using `watchOptions.followSymlinks` to the more conventional `config.symlinks` setting, and adds an alias for `react` to point to the one pulled in by the Superset application. This is necessary, as otherwise there will be two instances of React - one fro Superset, and one from `superset-ui`.

### TESTING INSTRUCTIONS
1. npm link a `superset-ui` package/plugin
2. make code changes to the package/plugin
3. watch the command line trigger the rebuild and the code changes reflected in the browser

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/webpack.config.js [hunk 1]
```diff
@@ -289,9 +289,10 @@ const config = {
         APP_DIR,
         './node_modules/@superset-ui/chart-controls',
       ),
+      react: path.resolve('./node_modules/react'),
```
Comment (kgabryje) [inline superset-frontend/webpack.config.js:292]: Perhaps we should have react as peer dep in superset-ui then?

Comment (villebro) [reply superset-frontend/webpack.config.js:292]: Hmm, that's a very good point. Let me try that.

File: superset-frontend/webpack.config.js [hunk 2]
```diff
@@ -289,9 +289,10 @@ const config = {
...
     },
     extensions: ['.ts', '.tsx', '.js', '.jsx', '.yml'],
-    symlinks: false,
+    symlinks: true,
```
Comment (kgabryje) [inline superset-frontend/webpack.config.js:295]: We can remove that line, `symlinks: true` is the default

File: superset-frontend/webpack.config.js (hunk 1 above)
Comment (villebro) [reply superset-frontend/webpack.config.js:292]: It apparently already is:
image

Comment (kgabryje) [reply superset-frontend/webpack.config.js:292]: Huh, then it's rather strange that it requires resolving

**Review by kgabryje — APPROVED** [review]
Thanks for fixing that pesky stuff! ❤️

**Pull request merged**